browser/src/view/components/Button/Button.js

```diff
@@ -1,40 +1,60 @@
 import React from 'react';
 import './Button.scss';
 
-export const Button = ({ label, style, icon }) => {
-  const renderButtonStyle = buttonStyle => {
-    switch (buttonStyle) {
-      case 'icon-only':
-        return icon;
-      case 'text-only':
-        return label;
-      case 'icon-text':
-        return (
-          <>
-            {icon} {label}
-          </>
-        );
-      default:
-        return null;
-    }
-  };
-  return <button className='btn-icon-container'>{renderButtonStyle(style)}</button>;
+export const Button = ({ layout, label, Icon, type }) => {
+  return (
+    <button
+      className={`${
+        type === ButtonAttributes.Type.destructive
+          ? 'btn-destructive'
+          : type === ButtonAttributes.Type.primary
+          ? 'btn-primary'
+          : 'btn-secondary'
+      }
+      ${
+        type === ButtonAttributes.Layout.iconAndLabel
+          ? 'btn-icon-label'
+          : type === ButtonAttributes.Layout.iconOnly
+          ? 'btn-icon-only'
+          : 'btn-label-only'
+      }`}
+    >
+      {layout === ButtonAttributes.Layout.iconOnly ||
+      layout === ButtonAttributes.Layout.iconAndLabel
+        ? Icon
+        : null}
+      {layout === ButtonAttributes.Layout.labelOnly ||
+      layout === ButtonAttributes.Layout.iconAndLabel
+        ? label
+        : null}
+    </button>
+  );
 };
 
-/*
-Button:
-- Style: icon only, text only, icon + text // type: normal, secondary, destructive
-- Interaction states: default, hover, selected (I don't think I'll worry about an :active/pressed button styling)
+const ButtonAttributes = {
```

Comment:
I think creating something referential may be helpful here.  

```
Button.ATTRIBUTES = { ... },
Button.TYPE = { ... }
```

This way it allows consumers to do something like:
```
<Button type={Button.TYPE.PRIMARY} />
```